src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_INITIAL_ARRAY_CAPACITY = 10;
+    private Object[] values;
+    private int size;
+    private int capacityOfArray;
+
+    public ArrayList() {
+        capacityOfArray = DEFAULT_INITIAL_ARRAY_CAPACITY;
+        values = new Object[capacityOfArray];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacityOfArray) {
+            growArray();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is invalid");
+        }
+        if (size() == capacityOfArray) {
+            growArray();
+        }
+        if (index == size()) {
+            values[size()] = value;
+            size++;
+        } else if (index < size()) {
+            Object[] newValuesOne = new Object[capacityOfArray];
+            Object[] newValuesTwo = new Object[capacityOfArray];
+            System.arraycopy(values, 0, newValuesOne, 0, index);
+            System.arraycopy(values, index, newValuesTwo, 0, capacityOfArray - index);
+            values = newValuesOne;
+            values[index] = value;
+            System.arraycopy(newValuesTwo, 0, values, index + 1, capacityOfArray - index - 1);
+            size++;
```

Comment:
could we use `System.arraycopy` method only ones here?

Reply:
one remained only